server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCasePreprocessorListener.java

```diff
@@ -230,6 +234,47 @@ public void exitParagraphStatement(ParagraphStatementContext ctx) {
                                     ctx.diagnostic()));
   }
 
+  @Override
+  public void enterFunctionDefinition(FunctionDefinitionContext ctx) {
+    push();
+  }
+
+  @Override
+  public void exitFunctionDefinition(FunctionDefinitionContext ctx) {
+    String multiTokenText = peek().toString().replaceAll("\\{\\$\\$\\*", "").replaceAll("}", "");
+    List<TerminalNode> word = ctx.multiToken().word().get(0).identifier().IDENTIFIER();
```

Comment:
word() could be empty.

Reply:
adjusted function declaration grammar. I am a bit reluctant to add  identifier rule (like FUNCTION-ID. func_id), as that might lead to adjusting other rule.

Reply:
WIP

Reply:
updated